client/jetpack-connect/remote-credentials.js

```diff
@@ -102,49 +106,69 @@ export class OrgCredentialsForm extends Component {
 	}
 
 	formFields() {
+		const { installError, isResponseCompleted, siteToConnect, translate } = this.props;
+		const { isSubmitting, password, username } = this.state;
+		const isFetching = ! installError && ! isResponseCompleted && isSubmitting && siteToConnect;
+
 		return (
 			<div>
-				<FormLabel htmlFor="username">{ this.props.translate( 'Username' ) }</FormLabel>
+				<FormLabel htmlFor="username">{ translate( 'Username' ) }</FormLabel>
 				<div className="jetpack-connect__site-address-container">
 					<Gridicon size={ 24 } icon="user" />
 					<FormTextInput
 						autoCapitalize="off"
 						autoCorrect="off"
 						className="jetpack-connect__credentials-form-input"
-						disabled={ this.state.submitting }
+						disabled={ isSubmitting }
 						id="username"
 						name="username"
 						onChange={ this.getChangeHandler( 'username' ) }
-						value={ this.state.username || '' }
+						value={ username || '' }
 					/>
 				</div>
 
 				<div className="jetpack-connect__password-container">
-					<FormLabel htmlFor="password">{ this.props.translate( 'Password' ) }</FormLabel>
+					<FormLabel htmlFor="password">{ translate( 'Password' ) }</FormLabel>
 					<div className="jetpack-connect__password-form">
 						<Gridicon size={ 24 } icon="lock" />
 						<FormPasswordInput
 							className="jetpack-connect__password-form-input"
-							disabled={ this.state.submitting }
+							disabled={ isSubmitting }
 							id="password"
 							name="password"
 							onChange={ this.getChangeHandler( 'password' ) }
-							value={ this.state.password || '' }
+							value={ password || '' }
 						/>
 					</div>
+					{ isFetching ? <Spinner /> : null }
 				</div>
 			</div>
 		);
 	}
 
+	renderButtonLabel() {
+		const { installError, isResponseCompleted, siteToConnect, translate } = this.props;
+		const { isSubmitting } = this.state;
+		const isFetching = ! installError && ! isResponseCompleted && isSubmitting && siteToConnect;
```

Comment:
We use this same logic twice, so you may want to make `isFetching` a prop and have this logic once inside `connect`?